### ISSUE TITLE
[Bug] Clicking to open the next volume opens the next volume at the last page

### ISSUE THREAD

**Issue by kanjieater** [gotson/komga#689]
### Komga environment
- **OS**:
- **Komga version**:
- [ ] **I am running Komga with Docker**
  - Docker image tag [e.g. latest, beta]:
- [X] **I am running Komga from the `jar`**
  - Java version:
- [X] **I have a problem in the web interface**
  - Browser (with version):
- [ ] **I have a problem with an OPDS client application**
  - OPDS Application (with version):
- [ ] **I have a problem with the Tachiyomi extension**
  - Tachiyomi version:
  - Tachiyomi extension version:

### Describe the bug
Clicking to open the next volume opens the next volume at the last page.

### Steps to reproduce
1. Click through to the end of a volume
2. Click once on the last page to get the popup about clicking again to go to the next book
3. Click to get to the next book
4. It opens the last page
If the book ahead of it has been opened before it won't do it. Not sure if it's related to right -> left reading mode, which i do have set for all books

### Expected behavior
Opening the next volume doesn't jump to the last page, it should open the first page of the next volume. 

### Actual behavior
Clicking to open the next volume opened the last page of the 2nd volume. Spoilers! T_T

### Additional context
N/A

### Screenshots
N/A

### Log file
If applicable, add an excerpt of the log file (max 20 lines) _AND_ attach the complete log file or a link to a gist/pastebin containing the log file ([where to find the logs](https://komga.org/faq/#where-can-i-find-the-log-files)).

**Comment by gotson** [comment]
Can you attach a video showing the problem, from the series page? 

> 4\. Not sure if it's related to right -> left reading mode, which i do have set for all books

Can you try and confirm behavior?

**Comment by gotson** [comment]
Please update the issue description too.

**Comment by kanjieater** [comment]
I am having difficulty preproducing it consistently. But I have had it happen multiple times.

**Comment by bmfrosty** [comment]
I'm seeing the same issue.  I identified it while using Komga under unraid which is acting as a docker orchestrator.  I don't know who's template I used, nor whether that matters.  The gui shows that I'm running v0.128.1-master.  I downloaded the files from the-eye - specifically something involving a man that was bitten by a spider, ultimately.

I had a theory for a bit that it starts happening in a session after opening cbz files or certain cbr files, but I don't know if that's correct.  I also suspected that once I opened one file that triggered the behaviour it would happen in every file I opened after that.  I don't know if any of that is correct.

**Comment by kanjieater** [comment]
Alright, I managed to capture how inconsistent it is too. https://imgur.com/a/REF7Jm4

It seems to have to do with page numbers. Notice in the gif, that if a book has MORE pages than the next book, it won't carry it over, if it has LESS it puts it near the end. Just try going through a book with more pages to less pages and you should be able to reproduce it.

**Comment by gotson** [comment]
I can't reproduce on my end. There is zero logging in the front-end at the moment, and that makes it impossible to troubleshoot javascript logic.

I am looking at some libraries to add logging in the front-end, so i could get useful logs from end-users when such problem arise, and be able to troubleshoot.

**Comment by gotson** [comment]
I have added basic logging in the webui for the web reader, around the places where i think the problem could be located.

Could you try to install v`0.128.2`? Then to enable the logs you will need to do the following:
- in the browser's developer console, type `localStorage.setItem('loglevel','DEBUG');`
- reload the page (F5)
- open a book for reading, you should see some output in the developer's console
- to save the logs, in Chrome you can right click in the console and `Save as...` to save to a file

Try to reproduce the problem with those logs enabled, and send me the logs alongside the book ID of the problematic book, so i can see what happened and when in the logs.

**Comment by kanjieater** [comment]
Here you go: [localhost-1632168926462.log](https://github.com/gotson/komga/files/7198504/localhost-1632168926462.log)


> I have added basic logging in the webui for the web reader, around the places where i think the problem could be located.
> 
> Could you try to install v`0.128.2`? Then to enable the logs you will need to do the following:
> 
> * in the browser's developer console, type `localStorage.setItem('loglevel','DEBUG');`
> * reload the page (F5)
> * open a book for reading, you should see some output in the developer's console
> * to save the logs, in Chrome you can right click in the console and `Save as...` to save to a file
> 
> Try to reproduce the problem with those logs enabled, and send me the logs alongside the book ID of the problematic book, so i can see what happened and when in the logs.

**Comment by gotson** [comment]
I might need more logs, because:

```
BookReader.vue:632 setup bookId:065MZ2BE9MNWQ, page:NaN
BookReader.vue:657 pages count: 193
BookReader.vue:658 read progress: null
BookReader.vue:739 goTo: 1 <===== this here
```

Clearly says it's opening page 1.

Let me skim through the code, see where i can slap more log statements. I'm quite happy we have some way to get logs on the front-end now 😄

**Comment by gotson** [comment]
more logs will be available in `0.128.3` if you can give it a try (when it's out)

**Comment by Daedren** [comment]
Here's my `0.128.3` logs.
I refreshed the page on 06AXYCPPK8R7C's last pages in Double Pages (no cover) mode, then went ahead to the next book (06AXYCPPK8R7D), which started on page 19-20 of 21 total

[daedren-20210922.log](https://github.com/gotson/komga/files/7210899/daedren-20210922.log)

I believe it's related to the book in this case having a double-spread and me being on Double Pages (no cover) mode. 
It happens to me consistently. I can provide one of these books if needed.

**Comment by gotson** [comment]
@Daedren would you be able to send me the 2 books via Discord PM, so i can try to reproduce the situation?

**Comment by gotson** [comment]
I think i know what's happening:
- the reader has 2 parameters of interest in this case: the list of pages, and the page number
- when setting up a new book (when you click next), 2 things happen:
  - the pages are changed
  - the page number is changed
- this is not an atomic change though, and what seems to be happening is:
  - the pages change, when that happens, the spreads are rebuilt
  - the new spreads array has a shorter length than the previous one, this triggers [this condition](https://github.com/gotson/komga/blob/cefd3f0ee8b24724ccaf41f00fdf382c0c74a513/komga-webui/src/components/readers/PagedReader.vue#L183), which changes the `currentpage`, and changes the page
  - the page is changed by the parent component (changing to 1), but that is ignored because the page has already changed beforehand

Now i need to think of how to fix it 😅

**Comment by gotson** [comment]
@Daedren no need to send me the books, I managed to reproduce on my end.

**Comment by gotson** [comment]
closed in 13760c5ef3117749966e90db9612674f5ab68b33
